src/v/cluster/archival_metadata_stm.h

```diff
@@ -52,7 +53,8 @@ class command_batch_builder {
     // command_batch_builder& operator=(command_batch_builder&&) = default;
     ~command_batch_builder() = default;
     /// Reset the manifest.
-    command_batch_builder& reset_metadata();
+    command_batch_builder&
+    reset_metadata(const cloud_storage::partition_manifest&);
```

Comment:
Just noticed that we had already recently introduced a `replace_manifest` stm command (see `command_batch_builder::replace_manifest` that does basically the same thing. May be worth just replicating that instead.